bundles/binding/org.openhab.binding.radius/src/main/java/org/openhab/binding/radius/internal/RadiusBinding.java

```diff
@@ -0,0 +1,202 @@
+
+	private static final Logger logger = LoggerFactory.getLogger(RadiusBinding.class);
+
+	/**
+	 * the refresh interval which is used to poll values from the Radius server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	/**
+	 * the address of the Radius servers (optional, defaults to 127.0.0.1)
+	 */
+	InetAddress radiusServer;
+
+	/**
+	 * the shared secret for the Radius server (required)
+	 */
+	private String sharedSecret = "";
+
+	public RadiusBinding() {
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void updated(Dictionary<String, ?> configuration) throws ConfigurationException {
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+		}
+
+		try {
+			String radiusServerAddress = (String) configuration.get("server");
+			if (StringUtils.isNotBlank(radiusServerAddress)) {
+				radiusServer = InetAddress.getByName(radiusServerAddress);
+			} else {
+				radiusServer = InetAddress.getByName("127.0.0.1");
+			}
+		} catch (UnknownHostException e) {
+			logger.warn("Could not determine RADIUS server address");
+		}
+
+		sharedSecret = (String) configuration.get("sharedSecret");
+
+		if (StringUtils.isNotBlank(sharedSecret)) {
+			setProperlyConfigured(true);
+		} else {
+			setProperlyConfigured(false);
+		}
+
+		logger.debug("server={}, secret={}, refresh={}", radiusServer, sharedSecret, refreshInterval);
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "Radius Refresh Service";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+		for (RadiusBindingProvider provider : providers) {
+			for (String itemName : provider.getInBindingItemNames()) {
+				logger.debug("Item '{}' is about to be refreshed", itemName);
+				RadiusPacket p = new RadiusPacket(provider.getRadiusType(itemName));
+				RadiusPacket pRet = null;
+				p.addAttribute(new RadiusAVP(RadiusAttribute.USER_NAME, provider.getUserName(itemName)));
+				p.encodePapPassword(provider.getPassword(itemName), sharedSecret);
+				try {
+					DatagramSocket clientSocket = new DatagramSocket();
+
+					byte[] receiveData = new byte[1024];
+					DatagramPacket sendPacket = new DatagramPacket(p.getRawPacket(), p.getRawPacketLength(),
+							radiusServer, 1812);
+					logger.trace("Sent: {}", p);
+					clientSocket.send(sendPacket);
+					DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
+					clientSocket.receive(receivePacket);
+					clientSocket.close();
+					pRet = new RadiusPacket(receivePacket.getData());
+					logger.trace("Received: {}", pRet);
+				} catch (IOException e) {
+					logger.warn("Error while sending packet to server");
+				}
```

Comment:
refactor and move this try-catch into its own method (for the sake of readability)